### PR TITLE
fix: avoid contextBridge crash when RenderFrame address is reused

### PR DESCRIPTION
**Pull request by loc**
#### Description of Change
Fix a crash that was happening because 
a) the RenderFramePersistenceStore map was keyed on a pointer, which are not guaranteed to be unique in the process and
b) the RenderFramePersistanceStore was not removing itself from the map when it was released, causing a bad access exception.

h/t @nornagon for help debugging.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed contextBridge crash when opening and closing many windows.

### CODE REVIEW

File: shell/renderer/api/atom_api_context_bridge.cc [hunk 1]
```diff
@@ -47,20 +47,12 @@ content::RenderFrame* GetRenderFrame(const v8::Local<v8::Object>& value) {
...
-      content::RenderFrame*, context_bridge::RenderFramePersistenceStore*>>
-      store_map;
-  return *store_map;
-}
-
 context_bridge::RenderFramePersistenceStore* GetOrCreateStore(
     content::RenderFrame* render_frame) {
-  auto it = GetStoreMap().find(render_frame);
-  if (it == GetStoreMap().end()) {
+  auto it = context_bridge::GetStoreMap().find(render_frame->GetRoutingID());
```
Comment (deepak1556) [inline shell/renderer/api/atom_api_context_bridge.cc]: Will there be a possibility of cross process access to this function ? `GetRoutingID` is not unique across processes.

Comment (MarshallOfSound) [reply shell/renderer/api/atom_api_context_bridge.cc]: @deepak1556 No, the bridge operates across two contexts in the same process

Comment (deepak1556) [reply shell/renderer/api/atom_api_context_bridge.cc]: Thanks for confirming.

File: shell/renderer/api/context_bridge/render_frame_context_bridge_store.h [hunk 2]
```diff
@@ -58,11 +58,15 @@ class RenderFramePersistenceStore final : public content::RenderFrameObserver {
   // proxy maps are weak globals, i.e. these are not retained beyond
   // there normal JS lifetime.  You must check IsEmpty()
 
+  int32_t routing_id_;
```
Comment (ckerr) [inline shell/renderer/api/context_bridge/render_frame_context_bridge_store.h]: _(nit)_ This is set in the constructor and never changes, so might as well be `const`

**Pull request merged**